### PR TITLE
Fix #4764 UUID assertion logic & improve edge case coverage in NetworkUtilsTest

### LINKED ISSUE #4764: Improve unit test coverage and assertions in NetworkUtilsTest
The current implementation of NetworkUtilsTest contains a few issues that reduce the reliability and clarity of the tests.
1.Unreliable UUID assertion in testFilenameFromUrl
2.Missing edge case coverage

Particular area and files to be tested

- `core>utils>NetworkUtils`

Image

### PR DESCRIPTION
**Pull request by Piyush-Kumar-Mishra**
Fixes #4764

1. Previously the test cases that expected a generated UUID did not properly validate the UUID format. The test now verify that the returned value matches the standard UUID pattern using regex.
2. Added missing test cases
- when the url is empty
- when url ends with  /
- when url contains  /?

**Screenshots** 
image

### CODE REVIEW

**Comment by Piyush-Kumar-Mishra** [comment]
@MohitMaliFtechiz @kelson42 fixed the formatting issues.

File: core/src/test/java/org/kiwix/kiwixmobile/core/utils/NetworkUtilsTest.kt [hunk 1]
```diff
@@ -200,4 +200,42 @@ class NetworkUtilsTest {
...
+  }
+
+  @Test
+  fun `parseURL returns empty string when url is null`() {
+    val result = NetworkUtils.parseURL(context, null)
+    assertEquals("", result)
+  }
+
+  @Test
+  fun `parseURL returns empty string for url without underscore`() {
```
Comment (MohitMaliFtechiz) [inline core/src/test/java/org/kiwix/kiwixmobile/core/utils/NetworkUtilsTest.kt:249]: @Piyush-Kumar-Mishra Thank you for adding the test cases. But we should add some more scenarios. e.g. `query at beginning`, `handles malformed url`, etc.

These test will cover them.

```kotlin
@Test
  fun `getFileNameFromUrl handles query at beginning`() {
    val result = NetworkUtils.getFileNameFromUrl("?query=test")

    // Since there is no filename, UUID should be generated
    assertTrue(result.isNotEmpty())
  }

  @Test
  fun `parseURL returns empty when beginIndex greater than endIndex`() {
    every { context.getString(R.string.zim_no_pic) } returns "No Pictures"
    every { context.getString(R.string.zim_no_vid) } returns "No Videos"
    every { context.getString(R.string.zim_simple) } returns "Simple"

    val url = "http://example.com/file_a_2020.zim"

    val result = NetworkUtils.parseURL(context, url)

    assertEquals("", result)
  }

  @Test
  fun `parseURL handles malformed url and catches exception`() {
    every { context.getString(R.string.zim_no_pic) } returns "No Pictures"
    every { context.getString(R.string.zim_no_vid) } returns "No Videos"
    every { context.getString(R.string.zim_simple) } returns "Simple"

    val result = NetworkUtils.parseURL(context, "http://example.com/_")

    assertEquals("", result)
  }
```

**Comment by Piyush-Kumar-Mishra** [comment]
@MohitMaliFtechiz Added missing tests for edge cases.
